drivers/sensor/pixart/paa3905/paa3905_bus.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2025 Croxel Inc.
+ * Copyright (c) 2025 CogniPilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_PAA3905_BUS_H_
+#define ZEPHYR_DRIVERS_SENSOR_PAA3905_BUS_H_
+
+#include <stdint.h>
+#include <zephyr/rtio/rtio.h>
+
+#include "paa3905.h"
+#include "paa3905_reg.h"
+
+static inline int paa3905_bus_read(const struct device *dev,
```

Comment:
pass an optional callback in as a parameter if you want to allow for synchronous/asynchronous operation here, and you can make submit non-blocking perhaps

Reply:
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/86644#discussion_r1987863462 resolution